### DISCUSSION TITLE
The issue of SpeechBrain cache directory

### DISCUSSION THREAD

**Discussion by matakanobu** [pyannote/pyannote-audio#932] [Ideas]
Hi, I have the issue and idea regarding SpeechBrain cache directory.

The first time of loading SpeechBrain model,  i.e. when the SpeechBrain cache directory does not exist, there is no problem. 
But, after the second time, it seems that the cache directory has not been released, so switching to another SpeechBrain model is unavailable. 
So I clear the cache with like the following code before loading the model.

```python
from pyannote.audio.core.model import CACHE_DIR
for p in glob(f"{CACHE_DIR}/speechbrain/*"):
    os.remove(p)
```

Is it necessary to clear the cache before loading model? They are sym-links or downloaded from hub, so I think no problem with deleting them.

https://github.com/pyannote/pyannote-audio/blob/1dc198dba860602cbc54938ca79b714001f58357/pyannote/audio/pipelines/speaker_verification.py#L94-L98

**Comment by hbredin** [comment]
Would you mind providing me with a minimum reproducible example showing the _"switching to another SpeechBrain model is unavailable"_ problem. A link to a notebook on Google Colab would be ideal.

**Reply by hbredin** [reply]
Also, does this happen with SpeechBrain-only code?

**Reply by matakanobu** [reply]
I made a notebook.

https://colab.research.google.com/drive/1tR6li8dPegCZ3bZX_QWRcm4hNPjTYhYa?usp=sharing

> Also, does this happen with SpeechBrain-only code?

In diarization pipeline with SpeechBrain embedding.
As presented in the notebook, if first use HuggingFace model and then try to use an own trained model, it will not work.
It may happen with SpeechBrain-only code also.
